src/lib/Microsoft.Health.Extensions.Fhir.R4/Repository/FhirServiceRepository.cs

```diff
@@ -0,0 +1,129 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using IFhirClient = Microsoft.Health.Fhir.Client.IFhirClient;
+
+namespace Microsoft.Health.Extensions.Fhir.Repository
+{
+    public class FhirServiceRepository : IFhirServiceRepository
+    {
+        private readonly IFhirClient _fhirClient;
+
+        private readonly CancellationTokenSource _cancellationTokenSource;
```

Comment:
>_cancellationTokenSource [](http://example.com/codeflow?start=49&length=24)

Is there are reason to have a static cts?  I don't think this will behave properly in multi threaded scenarios.  This really should be an instance for each method.

Reply:
Updated to create an instance of new CancellationTokenSource() in each method